File: algo/26. Remove Duplicates from Sorted Array.py
# Two Pointers

class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        left = -1
        for i, num in enumerate(nums):
            if left >= 0:
                if num == nums[left]:
                    continue
            left += 1
            nums[left] = num
        return left + 1

# class Solution(object):
#     def removeDuplicates(self, nums):
#         """
#         :type nums: List[int]
#         :rtype: int
#         """
#         i = 0
#         while(i<len(nums)-1):
#             if nums[i] == nums[i+1]:
#                 nums.pop(i)
#             else:
#                 i += 1
#         return len(nums)

File: round2/26. Remove Duplicates from Sorted Array.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        if (nums.Length == 0) return 0;
        int size = 1;
        for (int index=1; index<nums.Length; index++) {
            if (nums[index] != nums[index-1]) {
                nums[size] = nums[index];
                size ++;
            }
        }
        return size;
    }
}